### ISSUE TITLE
Can we use LibLog for logging?

### ISSUE THREAD

**Issue by alastairs** [waives/waives.net#71]
We should revisit whether [LibLog](https://github.com/damianh/LibLog) is a good fit for us, to reduce the abstractions introduced by the SDK. During the development of #11, we decided to use a custom abstraction [per @markawightman75's comment](https://github.com/waives/waives.net/issues/11#issuecomment-414747107), but we think we might be able to get the benefits of LibLog (i.e., automatic and transparent detection of logging framework in use) without sacrificing the requirements of console-by-default and console-on-error.

**Comment by alastairs** [comment]
The results of my spike prove this is possible and desirable:

* There is no logging abstraction for us to expose and support, and no abstraction for consumers to implement
* All major logging providers are supported and transparently detected: the integration from the consumer's perspective is zero-config, it Just Works™ 🚀 
* There is no loss of functionality as regards default console logging: this is currently supported via an extension method which consumers need to call explicitly, and this will continue working as-is if we adopt LibLog.

There are some drawbacks:

* Testing of the logging output needs to be done via one of the supported logging libraries, and I couldn't run tests adequately against [Serilog's community-provided in-memory sink](https://github.com/sandermvanvliet/SerilogSinksInMemory): the shared logger instance meant we ended up with non-deterministic tests as they interfered with each other. The tests I've written in #71 use regex pattern matching on rendered console output to determine whether or not the correct log message was written, which is not as robust as the assertions provided by the in-memory sink. There is [a suggestion from LibLog's author](https://github.com/damianh/CapturingLogOutputWithXunit2AndParallelTests) to write a custom sink to work around the issues found with tests running in parallel against a shared static logger.
* Microsoft.Extensions.Logging is not currently supported by LibLog as a provider. This library provides an opinionated abstraction over logging libraries, so essentially fulfils the same purpose as LibLog. There are few available "sinks" for Microsoft.Extensions.Logging; I believe at least two of these (Console and VS Debug window) are provided as "getting started"-type components, and most real-world applications will use a more fully-featured library with the common abstraction provided by Microsoft. The good news is that an application doing this will still be supported by LibLog, as the underlying log provider will be detected and loaded as usual.

**Comment by alastairs** [comment]
We have agreed to implement logging via LibLog :+1: I'll open a new issue to track the development of that piece of work and close off this spike story.
